processor/src/parallel/core_trace_fragment/trace_row.rs

```diff
@@ -0,0 +1,604 @@
+//! Module which concerns itself with all the trace row building logic.
+
+use core::ops::ControlFlow;
+
+use miden_air::trace::{
+    CLK_COL_IDX, CTX_COL_IDX, DECODER_TRACE_OFFSET, FN_HASH_OFFSET, STACK_TRACE_OFFSET,
+    SYS_TRACE_WIDTH,
+    decoder::{
+        ADDR_COL_IDX, GROUP_COUNT_COL_IDX, HASHER_STATE_OFFSET, IN_SPAN_COL_IDX,
+        NUM_OP_BATCH_FLAGS, NUM_OP_BITS, NUM_USER_OP_HELPERS, OP_BATCH_FLAGS_OFFSET,
+        OP_BITS_EXTRA_COLS_OFFSET, OP_BITS_OFFSET, OP_INDEX_COL_IDX,
+    },
+    stack::{B0_COL_IDX, B1_COL_IDX, H0_COL_IDX, STACK_TOP_OFFSET, STACK_TOP_RANGE},
+};
+use miden_core::{
+    Felt, ONE, Operation, Word, ZERO,
+    mast::{
+        BasicBlockNode, CallNode, JoinNode, LoopNode, MastForest, MastNodeExt, OpBatch, SplitNode,
+    },
+};
+
+use super::CoreTraceFragmentFiller;
+use crate::{
+    decoder::block_stack::ExecutionContextInfo, parallel::core_trace_fragment::BasicBlockContext,
+    processor::StackInterface,
+};
+
+const HASH_CYCLE_LEN: Felt = Felt::new(miden_air::trace::chiplets::hasher::HASH_CYCLE_LEN as u64);
```

Comment:
This `HASH_CYCLE_LEN` constant appears in several places across the codebase:

- `processor/src/fast/execution_tracer.rs:37`
- `processor/src/decoder/mod.rs:48`
- `processor/src/parallel/core_trace_fragment/trace_row.rs:28` (this line)

Would it make sense to extract this into a shared constant or trait? It could help maintain consistency and avoid having the same value defined in multiple places.

Reply:
Done; created a `HASH_CYCLE_LEN_FELT` in `processor::utils`